### PR TITLE
Remove accelerator="ddp*" instances, replace with accelerator="gpu" and strategy flag

### LINKED ISSUE #1428: Fix Flash CI (special examples failing)

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

`special` tests are failing in the CI:

```python
>           raise ValueError(
                f"You selected an invalid accelerator name: `accelerator={accelerator!r}`."
                f" Available names are: {', '.join(self._accelerator_types)}."
            )
E           ValueError: You selected an invalid accelerator name: `accelerator='ddp'`. Available names are: cpu, cuda, hpu, ipu, mps, tpu.
```

This PR attempts to fix these issues by replacing `ddp_cpu` with `accelerator="cpu", strategy="ddp"` and `ddp_shared` with `strategy="ddp", accelerator="gpu"`.

Also, since some of our CI tests install PL 1.4.2 version, and PL 1.6.0+ take `strategy` argument for `ddp` and before that it used to be `accelerator` - this PR makes sure that an appropriate condition is added to check which version of PL is present in the CI.

Fixes #1428

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,7 +129,7 @@ model.serve()
 or make predictions from raw data directly.
 
 ```py
-trainer = Trainer(accelerator='ddp', gpus=2)
+trainer = Trainer(strategy='ddp', accelerator="gpu", gpus=2)
```
Comment (krshrimali) [inline README.md:132]:
```suggestion
trainer = Trainer(accelerator="gpu", gpus=2, strategy="ddp")
```

File: tests/image/classification/test_training_strategies.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def test_learn2learn_training_strategies_registry():
     assert TRAINING_STRATEGIES.available_keys() == ["anil", "default", "maml", "metaoptnet", "prototypicalnetworks"]
 
 
-def _test_learn2learning_training_strategies(gpus, accelerator, training_strategy, tmpdir):
+def _test_learn2learning_training_strategies(gpus, accelerator, training_strategy, trainer_strategy, tmpdir):
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:57]:
```suggestion
def _test_learn2learning_training_strategies(gpus, accelerator, training_strategy, tmpdir):
```

File: tests/image/classification/test_training_strategies.py [hunk 3]
```diff
@@ -87,15 +87,18 @@ def _test_learn2learning_training_strategies(gpus, accelerator, training_strateg
         training_strategy_kwargs={"ways": dm.num_classes, "shots": 4, "meta_batch_size": 4},
     )
 
-    trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator)
+    trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator, strategy=trainer_strategy)
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:90]:
```suggestion
trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator)
```

File: tests/image/classification/test_training_strategies.py [hunk 4]
```diff
@@ -87,15 +87,18 @@ def _test_learn2learning_training_strategies(gpus, accelerator, training_strateg
...
+
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:91]:
```suggestion
trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator)
```

File: tests/image/classification/test_training_strategies.py [hunk 5]
```diff
@@ -87,15 +87,18 @@ def _test_learn2learning_training_strategies(gpus, accelerator, training_strateg
...
 
 
 # 'metaoptnet' is not yet supported as it requires qpth as a dependency.
 @pytest.mark.parametrize("training_strategy", ["anil", "maml", "prototypicalnetworks"])
 @pytest.mark.skipif(not _LEARN2LEARN_AVAILABLE, reason="image and learn2learn libraries aren't installed.")
 def test_learn2learn_training_strategies(training_strategy, tmpdir):
-    _test_learn2learning_training_strategies(0, None, training_strategy, tmpdir)
+    _test_learn2learning_training_strategies(
+        gpus=0, accelerator=None, training_strategy=training_strategy, trainer_strategy=None, tmpdir=tmpdir
+    )
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:101]:
```suggestion
_test_learn2learning_training_strategies(0, None, training_strategy, tmpdir)
```

File: tests/image/classification/test_training_strategies.py [hunk 6]
```diff
@@ -111,4 +114,6 @@ def test_wrongly_specified_training_strategies():
 @pytest.mark.skipif(not os.getenv("FLASH_RUNNING_SPECIAL_TESTS", "0") == "1", reason="Should run with special test")
 @pytest.mark.skipif(not _LEARN2LEARN_AVAILABLE, reason="image and learn2learn libraries aren't installed.")
 def test_learn2learn_training_strategies_ddp(tmpdir):
-    _test_learn2learning_training_strategies(2, "ddp", "prototypicalnetworks", tmpdir)
+    _test_learn2learning_training_strategies(
+        gpus=2, accelerator="gpu", training_strategy="prototypicalnetworks", trainer_strategy="ddp", tmpdir=tmpdir
+    )
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:119]:
```suggestion
_test_learn2learning_training_strategies(2, "ddp", "prototypicalnetworks", tmpdir)
```

File: tests/image/classification/test_training_strategies.py [hunk 7]
```diff
@@ -87,15 +87,15 @@ def _test_learn2learning_training_strategies(gpus, accelerator, training_strateg
         training_strategy_kwargs={"ways": dm.num_classes, "shots": 4, "meta_batch_size": 4},
     )
 
-    trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator)
+trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator)
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:90]:
```suggestion
    trainer = Trainer(fast_dev_run=2, gpus=gpus, accelerator=accelerator)
```

File: tests/image/classification/test_training_strategies.py [hunk 8]
```diff
@@ -87,15 +87,15 @@ def _test_learn2learning_training_strategies(gpus, accelerator, training_strateg
...
     trainer.fit(model, datamodule=dm)
 
 
 # 'metaoptnet' is not yet supported as it requires qpth as a dependency.
 @pytest.mark.parametrize("training_strategy", ["anil", "maml", "prototypicalnetworks"])
 @pytest.mark.skipif(not _LEARN2LEARN_AVAILABLE, reason="image and learn2learn libraries aren't installed.")
 def test_learn2learn_training_strategies(training_strategy, tmpdir):
-    _test_learn2learning_training_strategies(0, None, training_strategy, tmpdir)
+_test_learn2learning_training_strategies(0, None, training_strategy, tmpdir)
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:98]:
```suggestion
    _test_learn2learning_training_strategies(0, None, training_strategy, tmpdir)
```

File: tests/image/classification/test_training_strategies.py [hunk 9]
```diff
@@ -111,4 +111,4 @@ def test_wrongly_specified_training_strategies():
 @pytest.mark.skipif(not os.getenv("FLASH_RUNNING_SPECIAL_TESTS", "0") == "1", reason="Should run with special test")
 @pytest.mark.skipif(not _LEARN2LEARN_AVAILABLE, reason="image and learn2learn libraries aren't installed.")
 def test_learn2learn_training_strategies_ddp(tmpdir):
-    _test_learn2learning_training_strategies(2, "ddp", "prototypicalnetworks", tmpdir)
+_test_learn2learning_training_strategies(2, "ddp", "prototypicalnetworks", tmpdir)
```
Comment (krshrimali) [inline tests/image/classification/test_training_strategies.py:114]:
```suggestion
    _test_learn2learning_training_strategies(2, "ddp", "prototypicalnetworks", tmpdir)
```

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Pull request merged**